packages/backend/jobs-manager/service/src/modules/database/custom-jobs/jobs.provider.ts

```diff
@@ -0,0 +1,67 @@
+import { Logger } from '@nestjs/common';
+import { getModelToken } from '@nestjs/mongoose';
+import { UpdateFilter } from 'mongodb';
+import { Model } from 'mongoose';
+import { DataSources } from '../../datasources/data-sources';
+import { JobModelUpdateQueue } from '../../job-queue/job-model-update-queue';
+import { DATABASE_INIT } from '../admin/config/config.provider';
+import { JobPodConfiguration } from '../admin/config/job-pod-config/job-pod-config.model';
+import { CustomJobEntry } from './custom-jobs.model';
+import { GitJobSource, JobSource, JobSourceConfig } from './jobs.source';
+
+export const JOBS_INIT = 'JOBS_INIT';
+
+export const jobsInitProvider = [
+  {
+    provide: JOBS_INIT,
+    inject: [
+      getModelToken('customJobs'),
+      getModelToken('jobPodConfig'),
+      JobModelUpdateQueue,
+      DataSources,
+      { token: DATABASE_INIT, optional: false },
+    ],
+    useFactory: async (
+      jobsModel: Model<CustomJobEntry>,
+      jpcModel: Model<JobPodConfiguration>,
+      jobCodeQueue: JobModelUpdateQueue,
+      dataSources: DataSources,
+    ) => {
+      const logger = new Logger('jobsInitProvider');
+
+      try {
+        const podConfigs = await jpcModel.find();
+
+        const sourceConfigs: JobSourceConfig[] =
+          process.env.DATA_SOURCES != null
+            ? JSON.parse(process.env.DATA_SOURCES) ?? undefined
+            : [];
+
+        const sources: JobSource[] = [];
+        for (const source of sourceConfigs) {
+          const dataSource = await dataSources.get(source);
+          sources.push(new GitJobSource(dataSource));
+        }
+
+        for (const source of sources) {
+          const importedJobs = await source.synchronize(podConfigs);
+          for (const job of importedJobs) {
+            const filter: UpdateFilter<CustomJobEntry> = {
+              name: job.name,
+              'source.repoUrl': job.source?.repoUrl,
+            };
```

Comment:
Je sais que je l'ai pas implémenté comme ça, j'avais hésité (mb), mais ça a probablement plus de sens de permettre d'avoir les deux branches en même temps? C'est plus de flexibilité pour pas plus cher? Auquel cas il faudrait filtrer sur la branch ici, je pense.

Dans tous les cas, j'avais toujours prévu t'en parler, so there you go.